### PR TITLE
update Makefile to use cargo sparse protocol in all cargo commands

### PR DESCRIPTION
**Pull request by ibeckermayer**

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]

**Comment by r0mant** [comment]
@ibeckermayer Just FYI this PR breaks "Build on MacOS" check which ran inside the merge queue: https://github.com/gravitational/teleport/actions/runs/4570063117/jobs/8066974530.

Looks like we need to add top-level Makefile to path filters for test and build checks as well. For now I've removed your PR from the merge queue so it doesn't block other PRs from merging.

**Comment by zmb3** [comment]
@r0mant this is interesting, the "Build on macOS" check doesn't use the same version of Rust that our releases do, it just relies on whatever version of Rust is installed on the GitHub runners. We should probably fix that especially if we'll be using GHA for releases soon.

**Comment by ibeckermayer** [comment]
> Yeah, we do set Rust version in the build/release workflow: https://github.com/gravitational/teleport.e/blob/master/.github/workflows/build-mac-amd64.yaml#L99.

@r0mant @zmb3 I tried abstracting that section out into an action in `teleport.e` and use it both here and in the linked build-mac-amd64.yaml, but kept hitting roadblocks. I vaguely recalled hearing that our open source actions don't have access to the `teleport.e` repo, so assuming that's true then this copy-paste is the best we have for now.

If we want to create shared actions in the future, I think we'll need to create a separate actions repo that's available to both CI jobs as demonstrated here: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

File: Makefile [hunk 1]
```diff
@@ -79,6 +79,11 @@ endif
 CHECK_CARGO := $(shell cargo --version 2>/dev/null)
 CHECK_RUST := $(shell rustc --version 2>/dev/null)
 
+# Have cargo use sparse crates.io protocol:
+# https://blog.rust-lang.org/2023/03/09/Rust-1.68.0.html
+# TODO: Delete when it becomes default in Rust 1.70.0
+export CARGO_REGISTRIES_CRATES_IO_PROTOCOL=sparse
```
[the remark this answers is not in the document]
Comment (ibeckermayer) [reply Makefile:85]: @Tener the issue with using `CARGO_UNSTABLE_SPARSE_REGISTRY` is that it's only registered by a nightly build of `cargo`. So it has no effect using stable cargo of < 168.0.

**Pull request merged**